### PR TITLE
Add builder for pipeline notify and agents properties

### PR DESCRIPTION
**Pull request by jameslnewell**

### CODE REVIEW

File: src/schema/index.ts [hunk 1]
```diff
@@ -39,4 +40,7 @@ export type StepDependsOn = Extract<
 > extends (infer U)[]
   ? U
   : never;
+
 export type Field = Fields extends (infer U)[] ? U : never;
+
+export type PipelineNotify = BuildNotify extends (infer U)[] ? U : never;
```
Comment (jameslnewell) [inline src/schema/index.ts:46]: 🤔 I'm not super keen on exposing all these, not sure what to do tho, maybe I should create an internal types file??

Comment (johnjcsmith) [reply src/schema/index.ts:46]: What's the disadvantage in exporting these? Are they likely to be used?

File: src/builders/Pipeline.ts [hunk 2]
```diff
@@ -1,23 +1,51 @@
...
 export class Pipeline implements PipelineBuilder, StepsBuilder {
+  #agents: AgentsObject = {};
+  #notify: Array<PipelineNotify> = [];
   #steps: Array<StepSchema | StepBuilder> = [];
 
+  /**
+   * @see https://buildkite.com/docs/agent/v3/cli-start#agent-targeting
+   */
+  agent(tag: string, value: string): this {
+    this.#agents[tag] = value;
```
Comment (johnjcsmith) [inline src/builders/Pipeline.ts:17]: Should this be a helper as we will need it for CommandStep also?

File: src/schema/index.ts (hunk 1 above)
Comment (jameslnewell) [reply src/schema/index.ts:46]: Probably not very frequently used but I'm more worried that they might change frequently based off the JSON schema and the naming of these isn't particularly consistent.

**Review by johnjcsmith — APPROVED** [review]

**Pull request merged**